src/views/workflow-summary-tab/workflow-summary-tab.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import formatPayload from '@/views/workflow-page/helpers/format-payload';
+import { cssStyles } from './workflow-summary-tab.styles';
+import type { WorkflowPageTabContentProps } from '@/views/workflow-page/workflow-page-tab-content/workflow-page-tab-content.types';
+import WorkflowSummaryTabJsonView from './workflow-summary-tab-json-view/workflow-summary-tab-json-view';
+import useStyletronClasses from '@/hooks/use-styletron-classes';
+import SectionLoadingIndicator from '@/components/section-loading-indicator/section-loading-indicator';
+
+export default function WorkflowSummaryTab({
+  params,
+}: WorkflowPageTabContentProps) {
+  const { cls } = useStyletronClasses(cssStyles);
+  const [{ loading, error, data: workflowHistory }, setWorkflowHistory] =
```

Comment:
Maybe as a follow-up: would it be possible to make a hook/util that encapsulates this?
Or we could use something like react-query

Reply:
Still reading more about the available options.
The main point of research right now is how to sync cache if we are going to use multiple ways of fetching (RSC/Client).